homeassistant/components/philips_js/config_flow.py

```diff
@@ -0,0 +1,90 @@
+"""Config flow for Philips TV integration."""
+import logging
+from typing import Optional, TypedDict
+
+from haphilipsjs import ConnectionFailure, PhilipsTV
+import voluptuous as vol
+
+from homeassistant import config_entries, core, exceptions
+from homeassistant.const import CONF_API_VERSION, CONF_HOST
+
+from .const import DOMAIN  # pylint:disable=unused-import
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class FlowUserDict(TypedDict):
+    """Data for user step."""
+
+    host: str
+    api_version: int
+
+
+async def validate_input(hass: core.HomeAssistant, data: FlowUserDict):
+    """Validate the user input allows us to connect."""
+    hub = PhilipsTV(data[CONF_HOST], data[CONF_API_VERSION])
+
+    await hass.async_add_executor_job(hub.getSystem)
+
+    if hub.system is None:
+        raise ConnectionFailure
+
+    return hub.system
+
+
+class ConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for Philips TV."""
+
+    VERSION = 1
+    CONNECTION_CLASS = config_entries.CONN_CLASS_LOCAL_POLL
+
+    _default = {}
+
+    async def async_step_import(self, conf: dict):
+        """Import a configuration from config.yaml."""
+        for entry in self._async_current_entries():
+            if entry.data[CONF_HOST] == conf[CONF_HOST]:
+                return self.async_abort(reason="already_configured")
```

Comment:
Should this compare a serial number to be sure it refers to the same thing?

Reply:
That would require connecting to the device on every startup.

Reply:
On every import. You can choose to just import once.

Reply:
Yes. This what this blocks. Only import once.